src/index.js

```diff
@@ -0,0 +1,284 @@
+   * of an already exist network in the Status app.
+   * @private
+   * 
+   * @param {*} cb Callback called after the connect to network command 
+   * is sent to the Status app, called with parameters:
+   *  - {String} err Error that occurred adding and connecting a network.
+   *  - {Object} result Result returned from the Status app on successful 
+   *             connection to the Embark network.
+   * 
+   * @returns {void}
+   */
+  [_addAndConnectStatusNetwork](cb) {
+    // defined here because the configs have not been hydrated from the dapp when the plugin is instantiated
+    const nodePort = this.blockchainConfig.proxy ? this.blockchainConfig.rpcPort + 10 : this.blockchainConfig.rpcPort;
+    const nodeUrl = buildUrl('http', this.blockchainConfig.rpcHost, nodePort);
+    const networkName = `${NETWORK_NAME} (${this.pluginConfig.name})`;
+
+    this.logger.info(`Adding network '${networkName}' to Status...`);
+    return this.statusApi.addNetwork(networkName, nodeUrl, CHAIN_NAME, this.networkId, (err, result) => {
+      if (err) {
+        if (err.code === 'ECONNREFUSED' || err.code === 'ETIMEDOUT') {
+          return cb('Failed to add a network to the Status app. Is the Status app open?');
+        }
+        return cb(`Error while adding network ${nodeUrl} (name: ${CHAIN_NAME}, networkId: ${this.networkId}) to the Status App: ${err.message}.`);
+      }
+      if (!err && !result) {
+        return cb(`Empty response from the Status app. Is it possible your phone is in standby or the Status app is in the background?`);
+      }
+      const statusNetworkId = result['network-id'];
+      if (!statusNetworkId) {
+        return cb(`Status app returned a bad response (no 'network-id' present), could not connect.`);
+      }
+      this.logger.info(`Network '${networkName}' added successfully.`);
+      this[_setStatusNetworkId](statusNetworkId);
+      this[_connectToStatusNetwork](statusNetworkId, cb);
+    });
+  }
+
+  /**
+   * Retrieves the stored network ID of a previously added network. The network
+   * ID is stored in the dApp's temporary storage in the filesystem.
+   * @private
+   * 
+   * @param {*} cb Callback called after an attempt to read the network ID from
+   * the filesystem, called with parameters:
+   *  - {Error} err Error that occurred reading the file from the filesystem.
+   *  - {String | Buffer} data Contents of the stored networkId file.
+   * 
+   * @returns {void}
+   */
+  [_getStatusNetworkId](cb) {
+    fs.readFile(this.networkIdPath, cb);
+  }
+
+  /**
+   * Stores the network ID of the Status network in the temporary directory of the
+   * dApp in the filesystem.
+   * @private
+   * 
+   * @param {String} networkId ID of the Status network
+   * 
+   * @returns {void}
+   */
+  [_setStatusNetworkId](networkId) {
+    fs.mkdir(this.networkIdDirectory, _err => {
+      fs.writeFile(this.networkIdPath, networkId, err => {
```

Comment:
```suggestion
      fs.writeFile(this.networkIdPath, networkId, this.logger.error(`Error storing networkId in embark: ${err.message}`));
```
Again, would need to remove the other lines